filebeat/tests/system/test_crawler.py

```diff
@@ -837,3 +837,45 @@ def test_include_exclude_lines(self):
 
         # Check that output file has the same number of lines as the log file
         assert iterations == len(output)
+
+
+    def test_file_no_permission(self):
+        """
+        Checks that filebeat handles files without reading permission well
+        """
+
+        self.render_config_template(
+            path=os.path.abspath(self.working_dir) + "/log/*",
+        )
+        os.mkdir(self.working_dir + "/log/")
+
+        testfile = self.working_dir + "/log/test.log"
+        file = open(testfile, 'w')
+
+        iterations = 3
+        for n in range(0, iterations):
+            file.write("Hello World" + str(n))
+            file.write("\n")
+
+        file.close()
+
+        # Remove reading rights from file
+        os.chmod(testfile, 0o000)
+
+        if os.name == "nt":
+            current = stat.S_IMODE(os.lstat(testfile).st_mode)
```

Comment:
This currently doesn't work on Windows because of Note under https://docs.python.org/2/library/os.html#os.chmod

@andrewkroh Any idea how we could get this working?